XmlDocumentBundleTracker exception at startup

Using offline snapshot 426, I got this error at startup:
```
22:15:32.272 [ERROR] [ig.xml.osgi.XmlDocumentBundleTracker] - Could not send adding finished event for the module 'org.eclipse.smarthome.binding.hue'!
java.util.ConcurrentModificationException
        at java.util.HashMap$HashIterator.nextNode(HashMap.java:1429)[:1.8.0]
        at java.util.HashMap$EntryIterator.next(HashMap.java:1463)[:1.8.0]
        at java.util.HashMap$EntryIterator.next(HashMap.java:1461)[:1.8.0]
        at org.eclipse.smarthome.core.thing.xml.internal.XmlChannelTypeProvider.removeCachedChannelTypes(XmlChannelTypeProvider.java:382)[104:org.eclipse.smarthome.core.thing.xml:0.9.0.201607221840]
        at org.eclipse.smarthome.core.thing.xml.internal.XmlChannelTypeProvider.addChannelType(XmlChannelTypeProvider.java:130)[104:org.eclipse.smarthome.core.thing.xml:0.9.0.201607221840]
        at org.eclipse.smarthome.core.thing.xml.internal.ThingTypeXmlProvider.addingFinished(ThingTypeXmlProvider.java:135)[104:org.eclipse.smarthome.core.thing.xml:0.9.0.201607221840]
        at org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker.addingFinished(XmlDocumentBundleTracker.java:220)[96:org.eclipse.smarthome.config.xml:0.9.0.201607221840]
        at org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker.access$6(XmlDocumentBundleTracker.java:215)[96:org.eclipse.smarthome.config.xml:0.9.0.201607221840]
        at org.eclipse.smarthome.config.xml.osgi.XmlDocumentBundleTracker$1.processBundle(XmlDocumentBundleTracker.java:141)[96:org.eclipse.smarthome.config.xml:0.9.0.201607221840]
        at org.eclipse.smarthome.config.xml.osgi.AbstractAsyncBundleProcessor$1.run(AbstractAsyncBundleProcessor.java:231)[96:org.eclipse.smarthome.config.xml:0.9.0.201607221840]
        at java.lang.Thread.run(Thread.java:744)[:1.8.0]
```

--- a/bundles/core/org.eclipse.smarthome.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/XmlChannelTypeProvider.java
+++ b/bundles/core/org.eclipse.smarthome.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/XmlChannelTypeProvider.java
@@ -15,6 +15,7 @@ import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
 
 import org.eclipse.smarthome.core.common.osgi.ServiceBinder.Bind;
 import org.eclipse.smarthome.core.common.osgi.ServiceBinder.Unbind;
@@ -34,6 +35,7 @@ import org.osgi.framework.Bundle;
  * {@link XmlChannelTypeProvider} provides channel types from XML files.
  *
  * @author Dennis Nobel - Initial contribution
+ * @author Kai Kreuzer - fixed concurrency issues
  */
 public class XmlChannelTypeProvider implements ChannelTypeProvider {
 
@@ -98,8 +100,8 @@ public class XmlChannelTypeProvider implements ChannelTypeProvider {
 
     private Map<Bundle, List<ChannelType>> bundleChannelTypesMap;
 
-    private Map<LocalizedChannelTypeKey, ChannelGroupType> localizedChannelGroupTypeCache = new HashMap<>();
-    private Map<LocalizedChannelTypeKey, ChannelType> localizedChannelTypeCache = new HashMap<>();
+    private Map<LocalizedChannelTypeKey, ChannelGroupType> localizedChannelGroupTypeCache = new ConcurrentHashMap<>();
+    private Map<LocalizedChannelTypeKey, ChannelType> localizedChannelTypeCache = new ConcurrentHashMap<>();
 
     private ThingTypeI18nUtil thingTypeI18nUtil;
 

--- a/bundles/core/org.eclipse.smarthome.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/XmlThingTypeProvider.java
+++ b/bundles/core/org.eclipse.smarthome.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/XmlThingTypeProvider.java
@@ -15,6 +15,7 @@ import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
 
 import org.eclipse.smarthome.core.common.osgi.ServiceBinder.Bind;
 import org.eclipse.smarthome.core.common.osgi.ServiceBinder.Unbind;
@@ -37,6 +38,7 @@ import org.osgi.framework.Bundle;
  * @author Michael Grammling - Initial Contribution
  * @author Dennis Nobel - Added locale support, Added cache for localized thing types
  * @author Ivan Iliev - Added support for system wide channel types
+ * @author Kai Kreuzer - fixed concurrency issues
  */
 public class XmlThingTypeProvider implements ThingTypeProvider {
 
@@ -61,25 +63,33 @@ public class XmlThingTypeProvider implements ThingTypeProvider {
 
         @Override
         public boolean equals(Object obj) {
-            if (this == obj)
+            if (this == obj) {
                 return true;
-            if (obj == null)
+            }
+            if (obj == null) {
                 return false;
-            if (getClass() != obj.getClass())
+            }
+            if (getClass() != obj.getClass()) {
                 return false;
+            }
             LocalizedThingTypeKey other = (LocalizedThingTypeKey) obj;
-            if (!getOuterType().equals(other.getOuterType()))
+            if (!getOuterType().equals(other.getOuterType())) {
                 return false;
+            }
             if (locale == null) {
-                if (other.locale != null)
+                if (other.locale != null) {
                     return false;
-            } else if (!locale.equals(other.locale))
+                }
+            } else if (!locale.equals(other.locale)) {
                 return false;
+            }
             if (uid == null) {
-                if (other.uid != null)
+                if (other.uid != null) {
                     return false;
-            } else if (!uid.equals(other.uid))
+                }
+            } else if (!uid.equals(other.uid)) {
                 return false;
+            }
             return true;
         }
 
@@ -89,7 +99,7 @@ public class XmlThingTypeProvider implements ThingTypeProvider {
 
     }
 
-    private Map<LocalizedThingTypeKey, ThingType> localizedThingTypeCache = new HashMap<>();
+    private Map<LocalizedThingTypeKey, ThingType> localizedThingTypeCache = new ConcurrentHashMap<>();
 
     private Map<Bundle, List<ThingType>> bundleThingTypesMap;
 

